keps/sig-node/2837-pod-level-resource-spec/README.md

```diff
@@ -0,0 +1,1525 @@
+depending on the developer's activities.
+
+The proposed Pod Level Resource Spec feature addresses this by allowing all
+containers within the pod to share a defined resource pool. In the example
+provided, even though the IDE has its own resource constraints, tool1 and tool 2
+containers can dynamically utilize any remaining resources within the pod's
+overall limit. 
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  labels:
+    run: ide
+  name: myide
+spec:
+  resources:
+    limits:
+      memory: "1024Mi"
+      cpu: "4"
+  initContainers:
+    - image: shell
+      name: debian:buster
+    - image: tool1
+      name: first-tool:latest
+    - image: tool2
+      name: second-tool:latest
+  containers:
+    - name: ide
+      image: theia:latest
+      resources:
+        requests:
+          memory: 128M
+          cpu: "0.5"
+        limits:
+          memory: 256M
+          cpu: "1"
+```
+
+Without this feature, the developers will need to estimate beforehand how much
+resources to allocate to the tools, a task that can be challenging especially for
+workloads with varying resource demands.
+
+#### Better Resource Utilization
+
+In scenarios where multiple containers within a pod experience independent and
+uncoordinated resource usage spikes, the current container-level resource
+management can lead to significant resource wastage. This is because the
+resources are allocated based on each container's peak resource usage, even though
+the peaks might not occur simultaneously. Consequently, while one container is
+experiencing high resource usage, the other containers might be idle, resulting
+in underutilized resources within the pod.
+
+Pod Level Resources offer a solution for a more efficient resource allocation
+strategy. By setting a limit at pod level, resources can be shared dynamically
+among containers. This means that when one container experiences a spike, it can
+utilize the idle resources of the other containers within a pod. 
+
+Since resource bursts across containers are not synchronized, the aggregate
+resource usage of the pod, in this case, is generally lower than the sum of the
+individual container's peak usage. By setting the pod level limit to a percentile
+of this aggregate usage, we can achieve better resource utilization and avoid
+unnecessary over-allocation.
+
+### Notes/Constraints/Caveats
+
+1. cgroupv1 has been moved to maintenance mode since Kubernetes version 1.31.
+   Hence this feature will be supported only on cgroupv2.
+2. Pod Level Resource Specifications is an **opt-in** feature, and will have no
+effect on existing deployments. Only deployments that explicitly require this
+functionality should turn it on by setting the relevant resource section at
+pod-level in the Pod specification.
+
+### Risks and Mitigations
+
+1. Since Pod Level Resource Specifications is an **opt-in** feature, merging the
+   feature related changes won't impact existing workloads. Moreover, the feature
```

Comment:
same point as  below. If there are thrid party schedulers or other tools assuming how resources are being calculated, they may break. We have/had the same risk with the sidecar containers. Worth mentioning